gmso/external/convert_foyer.py

```diff
@@ -0,0 +1,582 @@
+from warnings import warn
+
+import numpy as np
+import unyt as u
+import os
+
+from lxml import etree
+from gmso.utils.io import has_foyer # Need for foyer
+
+if has_foyer:
+    import foyer
+
+def from_foyer(foyer_xml, gmso_xml):
+    """Convert a foyer XML to a gmso XML
+
+    Parameters
+    ----------
+    foyer_xml : XML
+        An XML file in the foyer format 
+   
+    Returns
+    -------
+    gmso_xml : XML
+        An XML file in the gmso format
+    """
+
+    foyer_xml_tree = etree.parse(foyer_xml)
+    f_kwargs = {
+            'atom_types': [],
+            'non_bonded_forces': [],
+            'harmonic_bond_types': [],
+            'harmonic_angle_types': [],
+            'urey_bradley_angle_types': [],
+            'rb_torsion_dihedral_types': [],
+            'periodic_torsion_dihedral_types': []
+            }
+
+    # Try to load in AtomType section
+    # Load in AtomTypes section otherwise
+    atom_types_el = foyer_xml_tree.findall('AtomType')
+    if len(atom_types_el) == 0:
+        atom_types_el = foyer_xml_tree.findall('AtomTypes')
```

Comment:
What happens if this is empty too? Or if it is a properly formatted foyer xml, but is empty?